src/gui/widgets/LcdFloatSpinBox.cpp

```diff
@@ -0,0 +1,255 @@
+	this->setLayout(outerLayout);
+}
+
+
+void LcdFloatSpinBox::update()
+{
+	const int whole = static_cast<int>(model()->value());
+	const float fraction = model()->value() - static_cast<int>(model()->value());
+	const int intFraction = fraction * std::pow(10.f, m_fractionDisplay.numDigits());
+	m_wholeDisplay.setValue(whole + m_displayOffset);
+	m_fractionDisplay.setValue(intFraction + m_displayOffset);
+
+	QWidget::update();
+}
+
+
+void LcdFloatSpinBox::contextMenuEvent(QContextMenuEvent* event)
+{
+	// for the case, the user clicked right while pressing left mouse-
+	// button, the context-menu appears while mouse-cursor is still hidden
+	// and it isn't shown again until user does something which causes
+	// an QApplication::restoreOverrideCursor()-call...
+	mouseReleaseEvent(nullptr);
+
+	CaptionMenu contextMenu(model()->displayName());
+	addDefaultActions(&contextMenu);
+	contextMenu.exec(QCursor::pos());
+}
+
+
+void LcdFloatSpinBox::mousePressEvent(QMouseEvent* event)
+{
+	if (event->button() == Qt::LeftButton &&
+		!(event->modifiers() & Qt::ControlModifier) &&
+		event->y() < m_wholeDisplay.cellHeight() + 2)
+	{
+		m_mouseMoving = true;
+		m_origMousePos = event->globalPos();
+		QApplication::setOverrideCursor(Qt::BlankCursor);
+
+		AutomatableModel *thisModel = model();
+		if (thisModel)
+		{
+			thisModel->addJournalCheckPoint();
+			thisModel->saveJournallingState(false);
+		}
+	}
+	else
+	{
+		FloatModelView::mousePressEvent(event);
+	}
+}
+
+
+void LcdFloatSpinBox::mouseMoveEvent(QMouseEvent* event)
+{
+	// switch between integer and fractional step based on cursor position
+	if (event->x() < m_wholeDisplay.width()) {m_intStep = true;}
+	else {m_intStep = false;}
+
+	if (m_mouseMoving)
+	{
+		int dy = event->globalY() - m_origMousePos.y();
+		if (gui->mainWindow()->isShiftPressed())
+			dy = qBound(-4, dy/4, 4);
+		if (dy > 1 || dy < -1)
+		{
+			model()->setInitValue(model()->value() - dy / 2 * getStep());
+			emit manualChange();
+			QCursor::setPos(m_origMousePos);
+		}
+	}
+}
+
+
+void LcdFloatSpinBox::mouseReleaseEvent(QMouseEvent*)
+{
+	if (m_mouseMoving)
+	{
+		model()->restoreJournallingState();
+
+		QCursor::setPos(m_origMousePos);
+		QApplication::restoreOverrideCursor();
+
+		m_mouseMoving = false;
+	}
+}
+
+
+void LcdFloatSpinBox::wheelEvent(QWheelEvent *event)
+{
+	// switch between integer and fractional step based on cursor position
+	if (event->x() < m_wholeDisplay.width()) {m_intStep = true;}
+	else {m_intStep = false;}
+
+	event->accept();
+	model()->setInitValue(model()->value() + ((event->delta() > 0) ? 1 : -1) * getStep());
+	emit manualChange();
+}
+
+
+void LcdFloatSpinBox::mouseDoubleClickEvent(QMouseEvent *)
+{
+	enterValue();
+}
+
+
+void LcdFloatSpinBox::enterValue()
+{
+	bool ok;
+	float new_val;
+
+	new_val = QInputDialog::getDouble(
+			this, tr("Set value"),
+			tr("Please enter a new value between %1 and %2:").
+			arg(model()->minValue()).
+			arg(model()->maxValue()),
```

Comment:
```suggestion
			tr("Please enter a new value between %1 and %2:").
				arg(model()->minValue()).
				arg(model()->maxValue()),
```

Just a suggestion to make it clearer that those are part of `tr` and not arguments to the `getDouble` call.